adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -351,11 +355,31 @@ protected virtual void OnAuthenticate()
         {
         }
 
+        /// <summary>
+        /// Some calls need to be made on the UI thread and this is the central place to check if we have an owner
+        /// window and if so, ensure we invoke on that proper thread.
+        /// </summary>
+        /// <param name="action"></param>
+        protected void InvokeHandlingOwnerWindow(Action action)
+        {
+            // We only support WindowsForms (since our dialog is winforms based)
```

Comment:
Will this work for a WPF application? i.e. is the ownerWindow a Control object?

Reply:
No, but our browser is only WinForms and the owner window can only be an IWin32Window or we fail.  We should investigate an option for embedding Edge which they now have as a WPF and WinForms plugin.